File: leetcode_ly/c++/HashTable/CountPrimes.cpp
class Solution {
public:
    int countPrimes(int n) {
        if(n <= 2){
            return 0;
        }
        bool* isPrimes = new bool[n];
        for(int i = 0; i < n; i++){
            isPrimes[i] = true;
        }
        
        for(int i = 3; i * i < n; i += 2){
            if(isPrimes[i]){
                for(int j = i * i; j < n; j = j + 2 * i){
                    isPrimes[j] = false;
                }
            }
        }
        int count = 1;
        for(int k = 3; k < n; k += 2){
            if(isPrimes[k]){
                count++;
            }
        }
        return count;
    }
};

File: leetcode_ly/java/Hash Table/CountPrimes.java
public class Solution {
    public int countPrimes(int n) {
        boolean[] f = new boolean[n];
        int count = 0;
        for(int i = 2; i < n; i++){
            if(!f[i]){
                count++;
                for(int j = 2; i * j < n; j++){
                    f[i * j] = true;
                }
            }
        }
        return count;
    }
}